include/mbedtls/check_config.h

```diff
@@ -66,6 +66,11 @@
 #error "MBEDTLS_HAVE_TIME_DATE without MBEDTLS_HAVE_TIME does not make sense"
 #endif
 
+#if defined(MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH) && \
+    !defined(MBEDTLS_CTR_DRBG_USE_128_BIT_KEY)
```

Comment:
Historically we haven't been very consistent about “sub-options”: if `MBEDTLS_FOO_SETTING` is an option that controls (mostly) `foo.c`, and `MBEDTLS_FOO_C` is disabled, is `MBEDTLS_FOO_SETTING` ignored (because the code it affects is fully disabled) or is setting it an error (because you can't have a partial foo without `foo.c`)?

Currently `MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` is ignored if `MBEDTLS_CTR_DRBG_C` is disabled, so it should continue this way.

Also, if `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` is defined, maybe `MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` should be automatically enabled? After all, if you've decided to only have AES-128, the fact AES-128 will be used rather than AES-256 wouldn't come as a surprise.

Reply:
Yes, I think the best way is @gilles-peskine-arm's suggestion: if `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` is defined, and `MBEDTLS_CTR_DRBG_C` is too, then `MBEDTLS_CTR_DRBG_USE_128_BIT_KEY` should be automatically enabled.

If we have `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` but `MBEDTLS_CTR_DRBG_C` isn't set, we don't care about `MBEDTLS_CTR_DRBG_USE_128_BIT_KEY`